pkg/pool-utils/test/BasePoolController.test.ts

```diff
@@ -0,0 +1,158 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import WeightedPool from '@balancer-labs/v2-helpers/src/models/pools/weighted/WeightedPool';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { WeightedPoolType } from '@balancer-labs/v2-helpers/src/models/pools/weighted/types';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { MONTH } from '@balancer-labs/v2-helpers/src/time';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { encodeInvestmentConfig } from '@balancer-labs/v2-asset-manager-utils/test/helpers/rebalance';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+
+const POOL_SWAP_FEE_PERCENTAGE = fp(0.01);
+const WEIGHTS = [fp(30), fp(60), fp(5), fp(5)];
+const PAUSE_WINDOW_DURATION = MONTH * 3;
+const BUFFER_PERIOD_DURATION = MONTH;
+const METADATA = '0x4b04c67fb743403d339729f8438ecad295a3a015ca144a0945bb6bb9abe3da20';
+
+let admin: SignerWithAddress;
+let other: SignerWithAddress;
+let assetManager: Contract;
+let pool: WeightedPool;
+let allTokens: TokenList;
+let vault: Vault;
+let poolController: Contract;
+
+before('setup signers', async () => {
+  [, admin, other] = await ethers.getSigners();
```

Comment:
Let's add a pool owner account, it feels odd to use the same account to control the authorizer and the controller.